packages/oid4vci-demo-frontend/src/components/Form/index.tsx

```diff
@@ -24,26 +25,26 @@ function getInitialState(form: DataFormRow[] | undefined): FormData {
     return transformFormConfigToEmptyObject(form)
 }
 
-const evaluateDefaultValue = (field: DataFormElement, payload: FormData): FormFieldValue => {
-    const payloadValue = payload[field.key]
-    if (payloadValue) {
-        return payloadValue
+const evaluateDefaultValue = (field: DataFormElement, formInitData: ImmutableRecord | undefined, formData: FormOutputData): FormFieldValue => {
+    const fieldValue = formData[field.key]
+    if (fieldValue) {
+        return fieldValue
     }
 
-    let defaultValue: FormFieldValue = field.defaultValue ?? ''
+    let defaultValue: FormFieldValue = formInitData?.[field.id] ?? field.defaultValue ?? ''
```

Comment:
and this code already shows the result of the change. it just makes things more complex and adds more if this then that erlse this/ that